android/modules/android/src/java/ti/modules/titanium/android/httpresponsecache/HttpResponseCacheModule.java

```diff
@@ -0,0 +1,170 @@
+package ti.modules.titanium.android.httpresponsecache;
+
+import org.appcelerator.kroll.KrollModule;
+import org.appcelerator.kroll.annotations.Kroll;
+import org.appcelerator.kroll.common.Log;
+import org.appcelerator.titanium.TiApplication;
+
+import ti.modules.titanium.android.AndroidModule;
+import android.app.Activity;
+import android.app.Notification;
+import android.net.http.HttpResponseCache;
+
+import java.io.File;
+import java.io.IOException;
+
+@Kroll.module(parentModule = AndroidModule.class)
+public class HttpResponseCacheModule extends KrollModule
+{
+	private static final String TAG = "HttpResponseCache";
+	private static final String CACHE_SIZE_KEY = "ti.android.cache.size.max";
+	private static final int DEFAULT_CACHE_SIZE = 25 * 1024; // 25MB
+
+	private String httpCachePath;
+	private long httpCacheSize;
+	private File httpCacheDir;
+
+	public HttpResponseCacheModule()
+	{
+		super();
+		httpCachePath = "http";
+		httpCacheSize =
+			TiApplication.getInstance().getAppProperties().getInt(CACHE_SIZE_KEY, DEFAULT_CACHE_SIZE) * 1024;
+	}
+
+	@Kroll.method
+	public boolean install() throws IOException
+	{
+		HttpResponseCache cache = HttpResponseCache.getInstalled();
+		if (cache != null) {
+			cache.flush();
+			cache.close();
+		}
+		TiApplication tiApp = TiApplication.getInstance();
+		File dir = tiApp.getApplicationContext().getExternalCacheDir();
```

Comment:
Why not have the `path` (or `httpCacheDir`) property value determine the directory *and* relative path to store the cache?

I'm worried here because the Android docs point out that if the requests contain private data, the external cache dir is a bad place to stick the cache due to access controls not being enforced on external storage.

I'm assuming it's because we don't actually expose the external cache dir anywhere in our APIs? cc @garymathews @jquick-axway

Reply:
>I'm assuming it's because we don't actually expose the external cache dir anywhere in our APIs?

Yes. Also this is how currently `cacheDir` calculated for [`TiResponseCache`](https://github.com/appcelerator/titanium_mobile/blob/7_2_X/android/titanium/src/java/org/appcelerator/titanium/TiApplication.java#L470).